org.osgi.test.assertj.cm/src/main/java/org/osgi/test/assertj/cm/configurationevent/ConfigurationEventConditions.java

```diff
@@ -0,0 +1,294 @@
+			"type is <" + ConfigurationEventType.toString(eventType) + ">", //
+			e -> " was <" + ConfigurationEventType.toString(e.getType()) + ">");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an <code>ConfigurationEvent.
+	 * Checking if the pid of the Event is not null.
+	 *
+	 * @return the condition
+	 */
+	public static Condition<ConfigurationEvent> pidNotNull() {
+
+		return new Condition<ConfigurationEvent>(e -> Objects.nonNull(e.getPid()), "pid is not <null>");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an <code>ConfigurationEvent.
+	 * Checking if the pid of the Event is null.
+	 *
+	 * @return the condition
+	 */
+	public static Condition<ConfigurationEvent> pidNull() {
+
+		return VerboseCondition.verboseCondition((ConfigurationEvent e) -> e.getPid() == null, "pid is <null>", //
+			e -> " was <" + e.getPid() + ">");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an <code>ConfigurationEvent.
+	 * Checking if the factoryPid of the Event is not null.
+	 *
+	 * @return the condition
+	 */
+	public static Condition<ConfigurationEvent> factoryPidNotNull() {
+
+		return new Condition<ConfigurationEvent>(e -> Objects.nonNull(e.getFactoryPid()), "factoryPid is not <null>");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an <code>ConfigurationEvent.
+	 * Checking if the factoryPid of the Event is null.
+	 *
+	 * @return the condition
+	 */
+	public static Condition<ConfigurationEvent> factoryPidNull() {
+
+		return VerboseCondition.verboseCondition((ConfigurationEvent e) -> e.getFactoryPid() == null,
+			"factoryPid is <null>", //
+			e -> " was <" + e.getFactoryPid() + ">");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an <code>ConfigurationEvent.
+	 * Checking if a given pid <b>matches</b> the pid of the Event.
+	 *
+	 * @param pid the pid
+	 * @return the condition
+	 */
+	public static Condition<ConfigurationEvent> pid(final String pid) {
+
+		return VerboseCondition.verboseCondition((ConfigurationEvent e) -> Objects.equals(e.getPid(), pid),
+			"pid equals <" + pid + ">", //
+			e -> " was <" + e.getPid() + ">");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an <code>ConfigurationEvent.
+	 * Checking if a given factoryPid <b>matches</b> the factoryPid of the
+	 * Event.
+	 *
+	 * @param factoryPid the factory pid
+	 * @return the condition
+	 */
+	public static Condition<ConfigurationEvent> factoryPid(final String factoryPid) {
+
+		return VerboseCondition.verboseCondition(
+			(ConfigurationEvent e) -> Objects.equals(e.getFactoryPid(), factoryPid),
+			"factoryPid equals <" + factoryPid + ">", //
+			e -> " was <" + e.getFactoryPid() + ">");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an <code>ConfigurationEvent.
+	 * Checking if the type <code>ConfigurationEvent.CM_DELETED</code>
+	 * <b>matches</b> the type of the Event.
+	 *
+	 * @return the condition
+	 */
+	public static Condition<ConfigurationEvent> typeDeleted() {
```

Comment:
We should have a single `type(int eventType)` method.

Reply:
removed